outport/process/executionOrder/transactionsExecutionOrder.go

```diff
@@ -64,30 +64,35 @@ func (s *sorter) PutExecutionOrderInTransactionPool(
 	header data.HeaderHandler,
 	body data.BodyHandler,
 	prevHeader data.HeaderHandler,
-) error {
+) ([]string, []string, error) {
 	blockBody, ok := body.(*block.Body)
 	if !ok {
 		log.Warn("s.PutExecutionOrderInTransactionPool cannot cast BodyHandler to *Body")
-		return nil
+		return nil, nil, nil
+	}
+
+	scheduledMbsFromPreviousBlock, err := s.mbsGetter.GetScheduledMBs(header, prevHeader)
+	if err != nil {
+		return nil, nil, err
 	}
 
 	// already sorted
 	transactionsToMe, scheduledTransactionsToMe, err := extractNormalTransactionAndScrsToMe(pool, blockBody, header)
 	if err != nil {
-		return err
+		return nil, nil, err
 	}
 
 	// need to be sorted
-	transactionsFromMe, scheduledTransactionsFromMe, err := s.extractNormalTransactionsAndInvalidFromMe(pool, blockBody, header, prevHeader)
+	transactionsFromMe, scheduledTransactionsFromMe, scheduledExecutedInvalidTxsHashesPrevBlock, err := s.extractNormalTransactionsAndInvalidFromMe(pool, blockBody, header, scheduledMbsFromPreviousBlock)
```

Comment:
I think that now we have normal, scheduled and invalid txs return by this method, so the name of it is not matching anymore it's behaviour. I suggest to use a simple name now like: extractTransactionsFromMe or something like this. Also you can use a DTO to encapsulate these 3 return values.

Reply:
added